addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -179,6 +179,8 @@ export class HtmlField extends Component {
                 powerboxCommands: [
                     {
                         category: this.env._t('Marketing Tools'),
+                        category_id: 'marketing_tools',
+                        command_id: 'dynamic_placeholder',
```

Comment:
Why do we need to duplicate all these data ? Can't you filter on `_t('Marketing Tools')` instead of filtering on `'Marketing Tools'` ?

Reply:
@dmo-odoo 
I'm not that familiar with the translation system, but I discussed this with Aurelien (awa), and it seems each module keeps its own traductions, in its own pot file, so the same wording could have multiple traductions in the same language in different modules, or something like that, so to avoid thinking about all that, we decided to add non-translated keys.

But tell me if I'm wrong and if we are certain that the comparison of translated content will always be the same ^^

Reply:
What is your use case exactly ?

Reply:
Currently we filter knowledge commands in `/template` and `/table` blocks, mostly because of the sizing of the blocks (does not make sense to have an embedded view in a cell in a table), and I think @awa-odoo has another task where he would add more filters on commands (but I don't have more details right now)

Reply:
To explain briefly my use case, we are trying to implement a "backend portal" the same way as project sharing, meaning portal users will have access to the backend form and editor for knowledge articles.

For them I would like to filter out the "/image" command, as they do not have access to ir.attachment model.
(At least that is the thinking for now, it could change in the future).

In any case, trying to do comparisons based on translated content does not seem very robust.
As you could filter out commands in a different module than the one of the editor, I'm guessing the .pot files will have separate entries and could have separate translations.

It does not hurt to be able to identify commands and categories based on a fixed constant.
But tell us if you think this is a bad idea :slightly_smiling_face:

Reply:
That's not necessarily a _bad_ idea per se, but since it introduces a lot of duplication, I'd rather be cautious about it and only do it as a last resort.

In particular, I feel like filtering on command names (or id for that matter) might not be the best approach.

- For the /image case, we already have an option which is `allowCommandImage` for exactly the same use case in other modules.

- For the contextual availability of some commands depending on where you are, it is possible to set a `isDisabled` callback on commands to disable them in certain contexts.

Let me know if the suggested alternatives does not suit your needs. :)

Reply:
@dmo-odoo Thanks for the info! `isDisabled` did not exist at the time I wrote the filters, now I know ;p